doc/source/whatsnew/v3.0.0.rst

```diff
@@ -31,7 +31,7 @@ Other enhancements
 - :func:`DataFrame.to_excel` now raises an ``UserWarning`` when the character count in a cell exceeds Excel's limitation of 32767 characters (:issue:`56954`)
 - :func:`read_stata` now returns ``datetime64`` resolutions better matching those natively stored in the stata format (:issue:`55642`)
 - Allow dictionaries to be passed to :meth:`pandas.Series.str.replace` via ``pat`` parameter (:issue:`51748`)
--
+- :func:`interval_range` Preserves start/end data types
```

Comment:
This should go under the `Interval` section

Reply:
Done!